app/Config/Mimes.php

```diff
@@ -509,11 +508,20 @@ public static function guessExtensionFromType(string $type, string $proposedExte
 
 		$proposedExtension = trim(strtolower($proposedExtension));
 
-		if ($proposedExtension !== '' && array_key_exists($proposedExtension, static::$mimes) && in_array($type, is_string(static::$mimes[$proposedExtension]) ? [static::$mimes[$proposedExtension]] : static::$mimes[$proposedExtension], true))
+		if ($proposedExtension !== '')
 		{
-			return $proposedExtension;
+				if(array_key_exists($proposedExtension, static::$mimes) && in_array($type, is_string(static::$mimes[$proposedExtension]) ? [static::$mimes[$proposedExtension]] : static::$mimes[$proposedExtension], true))
+				{
+					// The detected mime type matches with the proposed extension.
+					return $proposedExtension;
+				}
+			
+				// An extension was proposed, but the media type does not match the mime type list.
+				return null;
```

Comment:
You indent this a little too much.

Reply:
Thanks, I don't really get along well with Atom yet :D